scripts/install-nix-from-closure.sh

```diff
@@ -1,181 +1,700 @@
-    exit 1
-fi
+    if [ -e "$dest"/store ]; then
+
+        if [ ! -d "$dest"/store ]; then
+            error "
+
+    '$dest/store' exists and it's not a directory nor link to one.
+    "
+        fi
+
+        if [ ! -w "$dest"/store ]; then
+            error "
 
-mkdir -p $dest/store
+    Directory '$dest/store' exists, but is not writable for user '$USER'.
 
-printf "copying Nix to %s..." "${dest}/store" >&2
+    This could indicate that another user has already performed a single-user
+    installation of Nix on this system.
+
+    If you wish to enable multi-user support see:
+    https://nixos.org/nix/manual/#ssec-multi-user
+
+    To nevertheless do a single-user install for '$USER':
+    recursively set user '$USER' as owner for '$dest'/store directory.
+    "
+        fi
 
-for i in $(cd "$self/store" >/dev/null && echo ./*); do
-    printf "." >&2
-    i_tmp="$dest/store/$i.$$"
-    if [ -e "$i_tmp" ]; then
-        rm -rf "$i_tmp"
     fi
-    if ! [ -e "$dest/store/$i" ]; then
-        cp -Rp "$self/store/$i" "$i_tmp"
-        chmod -R a-w "$i_tmp"
-        chmod +w "$i_tmp"
-        mv "$i_tmp" "$dest/store/$i"
-        chmod -w "$dest/store/$i"
+
+    if [ -d "$dest" ]; then
+        print "Directory $dest already exists. Skipping creation of $dest"
     fi
-done
-echo "" >&2
 
-echo "initialising Nix database..." >&2
-if ! $nix/bin/nix-store --init; then
-    echo "$0: failed to initialize the Nix database" >&2
-    exit 1
-fi
+    }
 
-if ! "$nix/bin/nix-store" --load-db < "$self/.reginfo"; then
-    echo "$0: unable to register valid paths" >&2
-    exit 1
-fi
+    checkMode() {
+        # Determine if we should punt to the single-user installer or not
+        if [ "$(uname -s)" = "Darwin" ]; then
+            INSTALL_MODE=daemon
+        elif [ "$(uname -s)" = "Linux" ] && [ -e /run/systemd/system ]; then
+            INSTALL_MODE=daemon
+        else
+            INSTALL_MODE=no-daemon
+        fi
 
-. "$nix/etc/profile.d/nix.sh"
+        # Trivially handle the --daemon / --no-daemon options
+        if [ "x${1:-}" = "x--no-daemon" ]; then
+            INSTALL_MODE=no-daemon
+        elif [ "x${1:-}" = "x--daemon" ]; then
+            INSTALL_MODE=daemon
+        elif [ "x${1:-}" != "x" ]; then
+            error "
+    Nix Installer [--daemon|--no-daemon]
+      --daemon:    Force the installer to use the Daemon
+                   based installer, even though it may not
+                   work.
+    
+      --no-daemon: Force a no-daemon, single-user
+                   installation even when the preferred
+                   method is with the daemon.
+    "
+        exit
+        fi
 
-if ! "$nix/bin/nix-env" -i "$nix"; then
-    echo "$0: unable to install Nix into your default profile" >&2
-    exit 1
-fi
+        if [ "$INSTALL_MODE" = "daemon" ]; then
+            printf '\e[1;31mSwitching to the Daemon-based Installer\e[0m\n'
```

Comment:
This is some output change. Please specify what it is supposed to do or create a function which does that with a meaningful name.

Reply:
This was added merged to `master`, and got into script.

There are still parts of script that are old nondescriptive code, that I am rewriting in nearest days.

Sorry for inconsistency here and there in the code.